cmd/clusterctl/client/repository/repository_github_test.go

```diff
@@ -313,29 +314,25 @@ func Test_gitHubRepository_downloadFilesFromRelease(t *testing.T) {
 	}
 	for _, tt := range tests {
 		t.Run(tt.name, func(t *testing.T) {
-			g, err := newGitHubRepository(providerConfig, configVariablesClient)
-			if err != nil {
-				t.Fatal(err)
-			}
-			g.injectClient = client
+			gRepo, err := newGitHubRepository(providerConfig, configVariablesClient)
+			g.Expect(err).NotTo(HaveOccurred())
 
-			got, err := g.downloadFilesFromRelease(tt.args.release, tt.args.fileName)
-			if (err != nil) != tt.wantErr {
-				t.Fatalf("error = %v, wantErr %v", err, tt.wantErr)
-			}
+			gRepo.injectClient = client
+
+			got, err := gRepo.downloadFilesFromRelease(tt.args.release, tt.args.fileName)
 			if tt.wantErr {
+				g.Expect(err).To(HaveOccurred())
 				return
 			}
 
-			if !reflect.DeepEqual(got, tt.want) {
-				t.Errorf("got = %v, want %v", got, tt.want)
-			}
+			g.Expect(err).NotTo(HaveOccurred())
+			g.Expect(got).To(Equal(tt.want))
 		})
 	}
 }
 
 func testMethod(t *testing.T, r *http.Request, want string) {
-	if got := r.Method; got != want {
-		t.Errorf("Request method: %v, want %v", got, want)
-	}
+	g := NewWithT(t)
+
+	g.Expect(r.Method).To(Equal(want))
 }
```

Comment:
can we just remove this? seems un-necessary

Reply:
+1
This is testing the github client and not necessarily anything from clusterctl library.

Reply:
reverted @vincepri

Reply:
Oh I was more curious if we could substitute calls to `testMethod` with `g.Expect(r.Method).To(Equal(want))`

Reply:
i can open a PR to propose that @vincepri if you think that is useful